here-naksha-lib-view/src/main/java/com/here/naksha/lib/view/missing/ObligatoryLayerResolver.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2017-2023 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package com.here.naksha.lib.view.missing;
+
+import com.here.naksha.lib.core.models.storage.FeatureCodec;
+import com.here.naksha.lib.view.MissingIdResolver;
+import com.here.naksha.lib.view.ViewLayer;
+import com.here.naksha.lib.view.ViewLayerRow;
+import java.util.List;
+import org.apache.commons.lang3.tuple.Pair;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class ObligatoryLayerResolver<FEATURE, CODEC extends FeatureCodec<FEATURE, CODEC>>
+    implements MissingIdResolver<FEATURE, CODEC> {
+
+  private final ViewLayer obligatoryLayer;
+
+  public ObligatoryLayerResolver(@NotNull ViewLayer obligatoryLayer) {
+    this.obligatoryLayer = obligatoryLayer;
+  }
+
+  @Override
+  public boolean skip() {
+    return false;
+  }
+
+  @Override
+  public @Nullable Pair<ViewLayer, String> idsToSearch(@NotNull List<ViewLayerRow<FEATURE, CODEC>> multiFeature) {
+    boolean featureFromObligatoryLayerExists = multiFeature.stream()
+        .anyMatch(layerRow -> layerRow.getViewLayerRef().equals(obligatoryLayer));
+    if (featureFromObligatoryLayerExists || multiFeature.isEmpty()) {
+      return null;
+    }
+    return Pair.of(obligatoryLayer, multiFeature.get(0).getRow().getId());
```

Comment:
Logically I think, there may be a need to search missing Id in multiple layers (other than obligatoryLayer too). Not sure what is the scope of missing-id search agreed as per design?

Reply:
ok, I've change it from one obligatoryLayer to set of obligatory layers, so now it's possible to configure one, few or all. For all other custom cases client has to deliver own implementation, unless it's already known what might be useful